lib/galaxy/tools/wrappers.py

```diff
@@ -642,9 +642,9 @@ def __init__(
         self.collection = collection
 
         elements = collection.elements
-        element_instances = {}
+        element_instances: Dict[str, Union[DatasetCollectionWrapper, DatasetFilenameWrapper]] = {}
```

Comment:
Maybe a type alias for `Union[DatasetCollectionWrapper, DatasetFilenameWrapper]` named `ElementWrapper` will increase readability and reduce duplication here.

Reply:
Thanks for the good suggestion! I have added the `DatasetCollectionElementWrapper` type alias.